src/components/Idler.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import { PureComponent } from 'react'
+import { translate } from 'react-i18next'
+import { createStructuredSelector } from 'reselect'
+import { connect } from 'react-redux'
+import { hasPasswordSelector, autoLockTimeoutSelector } from 'reducers/settings'
+import debounce from 'lodash/debounce'
+import { lock } from 'reducers/application'
+
+type Props = {
+  autoLockTimeout: string,
+  hasPassword: boolean,
+  lock: Function
+}
+
+const mapStateToProps = createStructuredSelector({
+  autoLockTimeout: autoLockTimeoutSelector,
+  hasPassword: hasPasswordSelector
+})
+
+const mapDispatchToProps = {
+  lock
+}
+
+class Idler extends PureComponent<Props> {
+  constructor(){
+    super();
+    this.checkForAutoLock.bind(this)
+    this.idleTimeHandler.bind(this)
+  }
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.debounceOnChange)
+    window.addEventListener('mouseover', this.debounceOnChange)
+    setInterval(this.checkForAutoLock, 10000)
```

Comment:
Remove the listeners & clear the interval in `componentWillUnmount` (even if never unmounted for now)